core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -639,7 +646,7 @@ protected Scope visitDelete(Delete node, Optional<Scope> scope)
             Scope tableScope = analyzer.analyzeForUpdate(table, scope, UpdateKind.DELETE);
             node.getWhere().ifPresent(where -> analyzeWhere(node, tableScope, where));
 
-            analysis.setUpdateType("DELETE", tableName, Optional.of(table));
+            analysis.setUpdateType("DELETE", tableName, Optional.of(table), ImmutableList.of());
```

Comment:
Delete should not have a list of columns at all or `Optional.empty()` or have all columns.